pkg/chunked/compressor/compressor.go

```diff
@@ -119,8 +258,13 @@ func writeZstdChunkedStream(destFile io.Writer, outMetadata map[string]string, r
 
 		chunks := []chunk{}
 
+		hf := &holesFinder{
+			threshold: holesThreshold,
+			reader:    bufio.NewReader(tr),
+		}
+
 		rcReader := &rollingChecksumReader{
-			reader:  bufio.NewReader(tr),
+			reader:  hf,
```

Comment:
(Embedding `holesFinder` directly into `rollingChecksumReader`, without the pointer indirection, might help the compiler a bit. Or `ReadByte` could just be a private method of `rollingChecksumReader` — OTOH that might make testing it a bit harder.)